pyro/distributions/transforms/affine_coupling.py

```diff
@@ -53,13 +56,13 @@ class AffineCoupling(TransformModule):
 
     :param split_dim: Zero-indexed dimension :math:`d` upon which to perform input/output split for transformation.
     :type split_dim: int
-    :param hypernet: an autoregressive neural network whose forward call returns a real-valued
-        mean and logit-scale as a tuple. The input should have final dimension split_dim and the output final
-        dimension input_dim-split_dim for each member of the tuple.
-    :type hypernet: callable
-    :param log_scale_min_clip: The minimum value for clipping the log(scale) from the autoregressive NN
+    :param nn: a neural network whose forward call returns a real-valued mean and logit-scale as a tuple. The
```

Comment:
Changing a param name of a documented class is not backwards compatible, so you should hold off renaming until a 2.0 release that allows breaking changes. I think we're tentatively planning on a 2.0 release in 2021 including major language redesign. Keep this in mind when choosing public interfaces 😉

Reply:
Hehe sorry about that! I've switched it back. Good to know we can reconsider the design for a 2.0 release - I think we can make some really great changes then to consolidate the bijective transform library

Reply:
@fritzo - The attribute name is changed but the argument remains the same. That's fine by me since this maintains the current interface, or at least the extent to which I think we have committed. However if we consider attributes (those without leading underscores) to be part of the interface and if some code accesses it directly, it will break. Should we commit to keeping that backwards compatible too?

Reply:
I'm happy either way, provided there is freedom to change things around for a 2.0 release :)

Reply:
@neerajprad kwargs are part of the public interface, e.g. the proposed change would have broken
```py
transform = AffineCoupling(100, hypernet=MyMLP(100, 100))
```

Reply:
I agree that would be problematic. The case I was talking about would be someone using `transform.hypernet = ...` later, which would break with this change (they would have to change it to `transform.nn = ...`).